deeplake/core/dataset/dataset.py

```diff
@@ -1767,6 +1768,35 @@ def _delete_branch(self, name: str) -> None:
             self._set_read_only(read_only, err=True)
             self.storage.autoflush = self._initial_autoflush.pop()
 
+    @invalid_view_op
+    def squash_commits(self) -> None:
+        """
+        Squashes all commits in 'main' into one commit.
+        This is cannot be run if there are any branches besides `main`
+
+        Raises:
+            ReadOnlyModeError: If branch deletion is attempted in read-only mode.
+            VersionControlError: If the main branch cannot be squashed.
+        """
+        if self._is_filtered_view:
+            raise Exception(
+                "Cannot perform version control operations on a filtered dataset view."
+            )
+        read_only = self._read_only
+        if read_only:
+            raise ReadOnlyModeError()
+
+        try_flushing(self)
+
+        self._initial_autoflush.append(self.storage.autoflush)
+        self.storage.autoflush = False
+        try:
+            self._unlock()
+            squash_commits(self)
+        finally:
+            self._set_read_only(read_only, err=True)
```

Comment:
I guess here we also need to reset libdeeplake_dataset by `self.libdeeplake_dataset = None`

Reply:
Added it